src/HonzaBotner.Discord.Services/SlashCommands/BotSCommands.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Threading.Tasks;
+using DSharpPlus;
+using DSharpPlus.Entities;
+using DSharpPlus.SlashCommands;
+using DSharpPlus.SlashCommands.Attributes;
+using HonzaBotner.Discord.Services.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace HonzaBotner.Discord.Services.SlashCommands
+{
+    [SlashCommandGroup("bot", "Příkazy pro získání informací o botovi a jeho základní nastavení")]
+    public class BotSCommands : ApplicationCommandModule
+    {
+        private readonly ILogger<BotSCommands> _logger;
+        private readonly InfoOptions _options;
+
+        public BotSCommands(ILogger<BotSCommands> logger, IOptions<InfoOptions> options)
+        {
+            _logger = logger;
+            _options = options.Value;
+        }
+
+        [SlashCommand("activity", "Aktivita kterou bot zobrazí na svém profilu", false)]
+        [SlashRequireUserPermissions(Permissions.ManageGuild, false)]
+        public async Task ActivityAsync(InteractionContext ctx,
+            [Option("Aktivita", "Nazev vykonavane aktivity")] string message,
+            [Option("Druh", "Druh aktivity ktera se zobrazi u bota")]
+            [Choice("Competing", "Competing")]
+            [Choice("ListeningTo", "ListeningTo")]
+            [Choice("Playing", "Playing")]
+            [Choice("Watching", "Watching")]
+            string slashActivityType = "Competing")
+        {
+            ActivityType type = slashActivityType switch
+            {
+                "Competing" => ActivityType.Competing,
+                "ListeningTo" => ActivityType.ListeningTo,
+                "Playing" => ActivityType.Playing,
+                "Watching" => ActivityType.Watching,
+                _ => throw new ArgumentException("Unknown activity type", slashActivityType)
+            };
+
+            DiscordActivity activity = new (message, type);
+            try
+            {
+                await ctx.Client.UpdateStatusAsync(activity, UserStatus.Online);
+                DiscordInteractionResponseBuilder builder = new DiscordInteractionResponseBuilder()
+                    .WithContent($"Aktivita nastavena na {slashActivityType}: {message}");
+                await ctx.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning("Couldn't set activity to {SlashActivityType}: {Message}",
+                    slashActivityType, message);
+                DiscordInteractionResponseBuilder builder = new DiscordInteractionResponseBuilder()
+                    .WithContent($"Chyba: nepovedlo se nastavit aktivitu na {slashActivityType}:{message}")
+                    .AsEphemeral(true);
+                await ctx.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);
+            }
+        }
```

Comment:
Where (or how can it be done) we can allow this only for mods? 🤔 I see `[SlashRequireUserPermissions(Permissions.ManageGuild, false)]`, but that doesn't do anything, right?

Reply:
Mods as a role seems to be a bit harder to implement here than with Commands extension, looked into it a week ago, found it might be problematic, but I will look further into it.

Reply:
Well, we could also use some permissions-based system I guess, like if the user is admin or something.
But in some articles, I have seen that you could somehow use Discord roles.

Reply:
I lied... Even that method doesn't work.. just tested with another account. We have to wait.